ci/build_cpp.sh

```diff
@@ -16,9 +16,25 @@ rapids-logger "Begin cpp build"
 
 sccache --zero-stats
 
-# This calls mambabuild when boa is installed (as is the case in the CI images)
-RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rapids-conda-retry mambabuild conda/recipes/librmm
+RAPIDS_PACKAGE_VERSION=$(rapids-generate-version)
+export RAPIDS_PACKAGE_VERSION
+
+# --no-build-id allows for caching with `sccache`
+# more info is available at
+# https://rattler.build/latest/tips_and_tricks/#using-sccache-or-ccache-with-rattler-build
+rattler-build build --recipe conda/recipes/librmm \
+                    --experimental \
+                    --no-build-id \
+                    --channel-priority disabled \
+                    --output-dir "$RAPIDS_CONDA_BLD_OUTPUT_DIR" \
+                    -c rapidsai \
+                    -c rapidsai-nightly \
```

Comment:
I wonder if we actually need both of these channels or if we should tighten things up using [`rapids-is-release-build`](https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-is-release-build) to select one or the other.

Reply:
Agreed. We could probably use logic similar to https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-configure-conda-channels to define an environment variable with channels in https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-configure-rattler.

tl;dr We need to avoid using `rapidsai-nightly` or `dask/label/dev` when `rapids-is-release-build` returns true.